pkg/providers/tinkerbell/tinkerbell.go

```diff
@@ -263,19 +270,23 @@ func (p *Provider) DatacenterConfig(_ *cluster.Spec) providers.DatacenterConfig
 func (p *Provider) MachineConfigs(_ *cluster.Spec) []providers.MachineConfig {
 	configs := make(map[string]providers.MachineConfig, len(p.machineConfigs))
 	controlPlaneMachineName := p.clusterConfig.Spec.ControlPlaneConfiguration.MachineGroupRef.Name
-	p.machineConfigs[controlPlaneMachineName].Annotations = map[string]string{p.clusterConfig.ControlPlaneAnnotation(): "true"}
-	if p.clusterConfig.IsManaged() {
-		p.machineConfigs[controlPlaneMachineName].SetManagedBy(p.clusterConfig.ManagedBy())
+	if _, ok := p.machineConfigs[controlPlaneMachineName]; ok {
```

Comment:
I think we can revert this change. If these machine configs are missing, there's a catastrophic problem and silently ignoring the fact may result in side effects with processes later on.

The checks added in `NewProvider` should be sufficient to protect against this issue now.